### DISCUSSION TITLE
A small question about facet crease stiffness

### DISCUSSION THREAD

**Discussion by KaiXiao55** [amandaghassaei/OrigamiSimulator#38] [Q&A]
Dear author,

Thanks for creating this great simulator. I really enjoy playing it. 

While playing new structures, I found an interesting thing about facet crease stiffness in two different structures. And it looks like these two cases may lead to contradictive results.

Specifically, I tested a closed-loop structure (4-bar linkage) and an open-loop structure. While increasing the stiffness of facet-crease (I guess it increases the bending rigidity of face?) of both structures, the closed-loop structure's squared-shape face becomes non-flat whereas the open-loop one's face becomes flat, as demonstrated in the following figure.
![RCKQD688}ZM}1DFXQUQ)4ZD](https://user-images.githubusercontent.com/75075763/108295008-e3ddce80-71d1-11eb-9a2c-01073fa2e74f.png)

Are both results making sense? or my FOLD file has some problem?
[open_loop.txt](https://github.com/amandaghassaei/OrigamiSimulator/files/5999750/open_loop.txt)
[closed-loop.txt](https://github.com/amandaghassaei/OrigamiSimulator/files/5999751/closed-loop.txt)

I will very much appreciate it if you reply.

Kai Xiao.

**Comment by amandaghassaei** [comment]
Hi Kai,
Thanks for sending this, it helped me find a small bug in the fold importer where there was an issue with how angles were processed: https://github.com/amandaghassaei/OrigamiSimulator/commit/59b3b473200065ca3eca9ce1cd027c4bc9140a0e

You should modify your file to use a `null` angle when you want the m/v crease to behave like a hinge (this was broken before, but should work now) - the way you have it set now (at 0) will cause the simulator to drive the crease flat, which is not what you want (esp in the closed loop case).  Also you may want to try setting the assignment of the hinges to M instead of V in the closed loop case - the convention is kind of arbitrary, but I think that is what the sim is expecting.  You can always try setting the fold angle to -100% to check.

From what I can see, this was an issue with the way the fold angles on the m/v creases were being set, because they were set incorrectly (mostly my fault) they end up fighting with the facet creases in the closed loop case - leading to what you have in B.  So I think this is not actually an issue with the behavior of the facet creases.   I'm still not sure exactly how you got to A, but hopefully this issue is fixed now?

thanks for bringing this up!  we need to do a better job of testing out fold import - I think most people are using svgs - so this is very helpful.

**Reply by KaiXiao55** [reply]
Dear Amanda,

Thanks for your reply. I really appreciate it. 

By following your suggestions, I fixed my problems in assigning edges (using M instead of V) and using null value for the angles.  The results are getting better since the closed-loop structure can be fully folded now. (As shown in Fig. a to Fig. b when I set the foldAngle to 360). Looks like the problem is solved now. 

But if you don't mind, I still have a small question about the result shown in Fig. 1c and Fig .1d: when I set the foldAngle to 180 and the folding ratio equals 100%, the folding angle does not equal to 180 degrees as shown in Fig.d. 

Is the result in Fig. d reasonable? I thought the folding angle should reach its assigned value when the folding ratio is 100%.
![issue in 2021-03-14](https://user-images.githubusercontent.com/75075763/111071289-72fbaf00-8510-11eb-91cc-371af9b4d349.png)

Thanks a lot, 

Kai.

**Reply by amandaghassaei** [reply]
hmm - no it should get to the correct fold angle - can you attach the new files you're using?  In general the sim tries to get the fold angles to their target, but if other folds are interfering it can't guarantee that they will get there.  From what it looks like, it may be trying to fold the remaining "free" edges to 90 degrees, rather than letting them swing freely.  One thing to try. is to use the "strain visualization" mode in the app - that way you can see if folds are fighting against each other.  If there is no issue the structure will look dark blue, if there is a problems you will see greens/yellows maybe even reds.

Also if you are importing a fold file with null angles - be sure to uncheck the checkbox that says something like "use current angles as target angles" during the import - because that will overwrite your null values with whatever angle the creases are at when you import the file.   I should really make this all more clear - hopefully I'll have some time later this year for a redesign.  I suspect this is the current issue you're seeing.  Let me know if that fixes it!

**Reply by KaiXiao55** [reply]
Sorry for forgetting to attach the files I am using.  Now I have attached them. 

Following your suggestion with strain visualization, the result in fig. (a) exhibits dark blue faces since the folding stiffness is relatively low. however, it still can not reach a fully flat state even though folding ratio is 100%.  Besides, if fold stiffness is getting higher, looks like the folds are interfering as shown in Fig.(b).
![issues 03-15](https://user-images.githubusercontent.com/75075763/111116436-fd8bef00-85a0-11eb-8386-7e19409f72de.jpg)
[closed-loop_edges_foldAngle_180.txt](https://github.com/amandaghassaei/OrigamiSimulator/files/6139387/closed-loop_edges_foldAngle_180.txt)
[closed-loop_edges_foldAngle_360.txt](https://github.com/amandaghassaei/OrigamiSimulator/files/6139388/closed-loop_edges_foldAngle_360.txt)




As for the result where assigned foldAngle=-360, although the flat state looks reasonable (Fig. d), it also has an interfering state when folding percent=-100%. 

Looks like the folds are fighting against each other, at least in Fig. (a) and (b). How can I fix that? 

I also find the sim does not pop-ups checkbox of "use current angles as target angles" during import of those files. Is this normal?

Thanks a lot,

Kai
